File: js/storages/memoryStorage.js
const {data} = require("./data");

class MemoryStorage {
    Query(fn) {
        const results = [];

        let id = 1;
        for (const v of data) {
            if (fn(v)) {
                results.push({
                    id: id++,
                    name: v.name,
                    version: v.version,
                    tags: v.tags,
                    status: v.status,
                    createdAt: v.createdAt
                });
            }
        }

        return results;
    }
}

module.exports = { MemoryStorage };

File: go/storages/memory_storage.go
package storages

type MemoryStorage struct {
}

func (MemoryStorage) Query(fn func(map[string]interface{}) bool) ([]map[string]interface{}, error) {
	results := make([]map[string]interface{}, 0)
	for id, v := range data {
		if fn(map[string]interface{}{
			"name":      v.name,
			"version":   v.version,
			"tags":      v.tags,
			"status":    v.status,
			"createdAt": v.createdAt,
		}) {
			results = append(results, map[string]interface{}{
				"id":        id + 1,
				"name":      v.name,
				"version":   v.version,
				"tags":      v.tags,
				"status":    v.status,
				"createdAt": v.createdAt,
			})
		}
	}

	return results, nil
}
